src/gui/guiInventoryList.cpp

```diff
@@ -127,26 +128,29 @@ void GUIInventoryList::draw()
 								v2s32(x2 + border, y2)), clipping_rect_ptr);
 		}
 
-		// layer 1
-		if (selected)
-			item.takeItem(m_fs_menu->getSelectedAmount());
-
-		if (!item.empty()) {
-			// Draw item stack
+		// layer 1: draw item stack
+		// (avoid copying the object unless necessary here)
+		if (selected) {
+			ItemStack item2 = item;
+			item2.takeItem(m_fs_menu->getSelectedAmount());
+			if (item2.empty())
+				continue;
```

Comment:
well no, because `item2 != item`

Reply:
> well no, because `item2 != item`, so `item.empty() <=> item2.empty()` is not true

in practice it is though, it seems: https://github.com/minetest/minetest/pull/15603/files/934bb45c2336f14d7c26146e7c0498fa83884da5..09a9dc70e2d0c37e18d601e96b9b7484f6b65897#r1899832609